src/Mapbender/IntrospectionBundle/Command/SourcesCommand.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+
+namespace Mapbender\IntrospectionBundle\Command;
+
+
+use Mapbender\CoreBundle\Entity\Source;
+use Mapbender\CoreBundle\Entity\SourceInstance;
+use Mapbender\IntrospectionBundle\Component\Aggregator\Relation\ApplicationToSources;
+use Mapbender\IntrospectionBundle\Component\Aggregator\Relation\SourceToApplications;
+use Mapbender\IntrospectionBundle\Component\Aggregator\Relation\SourceToSourceInstances;
+use Mapbender\IntrospectionBundle\Component\Collector;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Formatter\OutputFormatterStyle;
+use Symfony\Component\Console\Helper\TableHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\HttpKernel\Kernel;
+
+class SourcesCommand extends ContainerAwareCommand
+{
+    protected $buckets = array();
+    protected $bucketBy = 'application';
+
+    protected function configure()
+    {
+        $this->setName('mapbender:inspect:source:usage');
+        $this->addOption('by-app', null, InputOption::VALUE_NONE, 'Group by application (default)');
+        $this->addOption('by-source', null, InputOption::VALUE_NONE, 'Group by source');
+        $this->addOption('unused-only', null, InputOption::VALUE_NONE, 'Display only unused sources');
+        $this->addOption('no-unused', null, InputOption::VALUE_NONE, 'Do not display unused sources');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('by-source')) {
+            if ($input->getOption('by-app')) {
+                throw new \RuntimeException("Options --by-app and --by-source are mutually exclusive!");
+            }
+            $this->bucketBy = 'source';
+        } else {
+            $this->bucketBy = 'application';
+        }
+        if ($input->getOption('unused-only') && $input->getOption('no-unused')) {
+            throw new \RuntimeException("Options --unused-only and --no-unused are mutually exclusive!");
+        }
+
+        $this->buckets = array();
+        $noteStyle = new OutputFormatterStyle('white', 'blue');
+        $output->getFormatter()->setStyle('note', $noteStyle);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $collector = new Collector($this->getContainer());
+        if ($this->bucketBy == 'application') {
```

Comment:
May be it's possible to divide this function for smaller ones to increase reading comfortability?

Reply:
Yes.

I also want to break the table formatting stuff out of that command, because it's used verbatim in other commands and should be in a central place.